students/shawn/class_03/mailroom.py

```diff
@@ -0,0 +1,256 @@
+                return get_user(new_donor[1].fnam + new_donor[1].lnam)
+
+# -----------------------------------------------------------------------*
+# Prompt user for arg
+# -----------------------------------------------------------------------*
+def send_thankyou():
+    """
+    If user exists, generates message in console.
+    :return: True
+    """
+    while True:
+        user=prompt_user("Enter the full name of the donor you with to thank (type 'list' to view donors)")
+        if user[0]:
+            if user[1].lower()=="list":
+                for i, v in donor_list.donors.items():
+                    print(v)
+                print("\n")
+            else:
+                if user[1].lower()=='quit':
+                    return True
+                else:
+                    donor=get_user(user[1])
+                    if donor != None:
+                        print(f"{'-'*100}"
+                              f"\n\nDearest {donor[1].fnam},\n\nThank for your latest donation of {donor[1].donation[-1]}, "
+                              f"bringing your total donations to {sum([i for i in donor[1].donation])}.\n\nSincerly,\nmanagment\n"
+                              f"{'-'*100}")
+
+                    return True
+
+
+# -----------------------------------------------------------------------*
+# Write report
+# -----------------------------------------------------------------------*
+def report(donors):
+    """
+    Calls the Donors.Report method to generate a report in the console
+    :param donors: global instance of Donors collection
+    :return: True
+    """
+    report=donors.report()
+    return True
+
+# -----------------------------------------------------------------------*
+# Main menu
+# -----------------------------------------------------------------------*
+def main_menu():
+    """
+    Entry point into app.
+    :return: Bool - true to continue, False to quit
+    """
+    msg="**Welcome to MailRoom**\nPlease select from the following actions\n\n\t(1)Send a thank you\n\t(2)Create a report\n\t(3)Quit\n\nSelect"
+    action=prompt_user(msg)
+    if action[0]:
+        if action[1] in ['1','2','3']:
+            if action[1]=='1':
+                ty=send_thankyou()
+                return ty
+            elif action[1]=='2':
+                rc=report(donor_list)
+                return rc
+            else:
+                print("Closing...")
+                return False
+
+
+if __name__ == '__main__':
+    # -----------------------------------------------------------------------*
+    # seed the database
+    # -----------------------------------------------------------------------*
+    donor_list=Donors()
+    load_default(donor_list)
+
+    # -----------------------------------------------------------------------*
+    # Prompt for action
+    # -----------------------------------------------------------------------*
+    status=True
+    while status==True:
+        status=main_menu()
+
+
```

Comment:
I like that you created a class here. Was there a reason you didn't create a class that encompasses the entire activity versus a class just for donors, etc?